AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/Support/MutationEventExtensionsTests.swift

```diff
@@ -0,0 +1,95 @@
+                                                        mutationType: .delete,
+                                                        version: nil)
+        let metadata = MutationSyncMetadata(id: modelId,
+                                            deleted: false,
+                                            lastChangedAt: Int(Date().timeIntervalSince1970),
+                                            version: 1)
+
+        let createMutationSync = MutationSync(model: post, syncMetadata: metadata)
+
+        let updateMutationExpectation = expectation(description: "save updateMutationEvent success")
+        storageAdapter.save(updateMutationEvent) { result in
+            guard case .success = result else {
+                XCTFail("Failed to save metadata")
+                return
+            }
+            updateMutationExpectation.fulfill()
+        }
+        wait(for: [updateMutationExpectation], timeout: 1)
+
+        let deleteMutationExpectation = expectation(description: "save deleteMutationEvent success")
+        storageAdapter.save(deleteMutationEvent) { result in
+            guard case .success = result else {
+                XCTFail("Failed to save metadata")
+                return
+            }
+            deleteMutationExpectation.fulfill()
+        }
+        wait(for: [deleteMutationExpectation], timeout: 1)
+
+        let queryAfterUpdatingVersionExpectation = expectation(description: "query success")
+        let updatingVersionExpectation = expectation(description: "update version success")
+        MutationEvent.updatePendingMutationEventVersionIfNil(for: post.id,
+                                                        mutationSync: createMutationSync,
+                                                        storageAdapter: storageAdapter) { result in
+            switch result {
+            case .failure(let error):
+                XCTFail("Error : \(error)")
+            case .success:
+                // query for head of mutation event table for given model id and check for version
+                updatingVersionExpectation.fulfill()
+                MutationEvent.pendingMutationEvents(for: post.id,
+                                                    storageAdapter: self.storageAdapter) { result in
+                    switch result {
+                    case .failure(let error):
+                        XCTFail("Error : \(error)")
+                    case .success(let mutationEvents):
+                        queryAfterUpdatingVersionExpectation.fulfill()
+                        guard !mutationEvents.isEmpty, let updatedEvent = mutationEvents.first else {
+                            XCTFail("Failure while updating version")
+                            return
+                        }
+                        XCTAssertEqual(updatedEvent.version, createMutationSync.syncMetadata.version)
+                        XCTAssertEqual(updatedEvent.mutationType, MutationEvent.MutationType.update.rawValue)
```

Comment:
my understanding is that the the fulfill call should be at the end, otherwise you will exit out of your test before checking because `wait(for: [queryAfterUpdatingVersionExpectation, updatingVersionExpectation], timeout: 1)` will be released when `queryAfterUpdatingVersionExpectation.fulfill()` is called and then 
```
guard !mutationEvents.isEmpty, let updatedEvent = mutationEvents.first else {
                            XCTFail("Failure while updating version")
                            return
                        }
                        XCTAssertEqual(updatedEvent.version, createMutationSync.syncMetadata.version)
                        XCTAssertEqual(updatedEvent.mutationType, MutationEvent.MutationType.update.rawValue)
``` 
won't be executed since test finishes